### PR TITLE
Add unit test for group score calc to assign replicas evenly

### LINKED ISSUE #4805: Scheduler: support the ability to automatically assign replicas evenly
**What would you like to be added**:
### Background
We want to introduce a new replica assignment strategy in the scheduler, which supports an even assignment of the target replicas across the currently selected clusters.

### Explanation
After going through the filtering, prioritization, and selection phases, three clusters(`member1`, `member2`, `member3`) were selected. We will automatically assign 9 replicas equally among these three clusters, the result we expect is `[{member1: 3}, {member2: 3}, {member3: 3}]`.

**Why is this needed**:
### User Story
As a developer, we have a deployment with 2 replicas that needs to be deployed with high availability across AZs. We hope Karmada can schedule it to two AZs and ensure that there is a replica on each AZ.
2AZ

Our PropagationPolicy might look like this:
```yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: foo
  namespace: bar
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
      name: foo
  placement:
    replicaScheduling:
      replicaSchedulingType: Divided
      replicaDivisionPreference: Weighted
      weightPreference:
        dynamicWeight: AvailableReplicas
    spreadConstraints:
    - spreadByField: zone
      maxGroups: 2
      minGroups: 2
    - spreadByField: cluster
      maxGroups: 2
      minGroups: 2
```
But unfortunately, the strategy `AvailableReplicas` does not guarantee that our replicas are evenly assigned.

### Any ideas?
We can introduce a new replica assignment strategy like `AvailableReplicas`, maybe we can name it `AverageReplicas`. It is essentially different from static weight assignment, because it does not support spread constraints and is mandatory. When assigning replicas, it does not consider whether the cluster can place so many replicas.

### PR DESCRIPTION
**Pull request by ipsum-0320**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Optimize the score calculation method to better measure the pros and cons of the group. This PR added unit test for group score calc to assign replicas evenly.

**Which issue(s) this PR fixes**:
Part of #5159 

**Special notes for your reviewer**:
@whitewindmills 

**Does this PR introduce a user-facing change?**:
```release-note
None.
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
Hi @ipsum-0320 Please rebase this PR and resolve the conflicts.

File: pkg/scheduler/core/spreadconstraint/group_clusters_test.go [hunk 1]
```diff
@@ -219,3 +220,162 @@ func Test_GroupClustersWithScore(t *testing.T) {
...
+		},
+	}
+	rbspecDynamicWeight := &workv1alpha2.ResourceBindingSpec{
+		Replicas: replica,
+		Placement: &policyv1alpha1.Placement{
+			ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+				ReplicaSchedulingType:     policyv1alpha1.ReplicaSchedulingTypeDivided,
+				ReplicaDivisionPreference: policyv1alpha1.ReplicaDivisionPreferenceWeighted,
+				WeightPreference: &policyv1alpha1.ClusterPreferences{
+					DynamicWeight: "AvailableReplicas",
```
Comment (whitewindmills) [inline pkg/scheduler/core/spreadconstraint/group_clusters_test.go:256]: use `policyv1alpha1.DynamicWeightByAvailableReplicas`?

File: pkg/scheduler/core/spreadconstraint/group_clusters_test.go [hunk 2]
```diff
@@ -219,3 +220,162 @@ func Test_GroupClustersWithScore(t *testing.T) {
...
+
+type GroupScoreArgs struct {
+	id          int
+	clusters1   []ClusterDetailInfo
+	clusters2   []ClusterDetailInfo
+	rbSpec      *workv1alpha2.ResourceBindingSpec
+	minGroups   int
+	want        bool
+	description string
+	// if clusters1 is better than clusters2, want is true
```
Comment (RainbowMango) [inline pkg/scheduler/core/spreadconstraint/group_clusters_test.go:301]: Is this the comment forgot to remove? Or put in the wrong place?

Comment (ipsum-0320) [reply pkg/scheduler/core/spreadconstraint/group_clusters_test.go:300]: I have put it in the correct place @RainbowMango

File: pkg/scheduler/core/spreadconstraint/group_clusters_test.go [hunk 3]
```diff
@@ -219,3 +220,162 @@ func Test_GroupClustersWithScore(t *testing.T) {
...
+type GroupScoreArgs struct {
+	id        int
+	clusters1 []ClusterDetailInfo
+	clusters2 []ClusterDetailInfo
+	rbSpec    *workv1alpha2.ResourceBindingSpec
+	minGroups int
+	want      bool
+	// if clusters1 is better than clusters2, want is true
+	description string
+}
```
Comment (RainbowMango) [inline pkg/scheduler/core/spreadconstraint/group_clusters_test.go:302]:
```suggestion
type GroupScoreArgs struct {
	id        int
	group1 []ClusterDetailInfo
	group2 []ClusterDetailInfo
	rbSpec    *workv1alpha2.ResourceBindingSpec
	minGroups int
	group1Wins      bool
	description string
}
```

Comment (ipsum-0320) [reply pkg/scheduler/core/spreadconstraint/group_clusters_test.go:302]: i think that's great. will done

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve
Thanks.

**Pull request merged**